include/fw_metadata.h

```diff
@@ -200,6 +203,42 @@ fw_firmware_info_get(u32_t firmware_address)
 }
 
 
+/* The supported offsets for the fw_info struct. */
+#define FW_INFO_OFFSET0 0x200
+#define FW_INFO_OFFSET1 0x400
+#define FW_INFO_OFFSET2 0x800
+#define FW_INFO_OFFSET_COUNT 3
+
+static const u32_t allowed_offsets[] = {FW_INFO_OFFSET0, FW_INFO_OFFSET1,
+					FW_INFO_OFFSET2};
+
+static_assert(ARRAY_SIZE(allowed_offsets) == FW_INFO_OFFSET_COUNT,
+		"Mismatch in the number of allowed offsets.");
+
+#if (FW_INFO_OFFSET_COUNT != 3) || \
```

Comment:
Why do we need this check when we have the static assert above?

Reply:
If offsets are added to the list, but this #if is not updated, this check will catch that.